variables.tf

```diff
@@ -1,4 +1,111 @@
-variable "example" {
-  description = "Example variable"
-  default     = "hello world"
+variable "account_enabled" {
```

Comment:
To me `account_enabled` tells me if the AWS account is enabled or not but I think this flag controls whether or not this module is enabled or disabled.

Could be misreading.

Reply:
Its part of the confusing aws structure of this, but the variable name might not be helping, It controls whether the Admin account is enabled or disabled for macie. however setting it to false and reapplying doesn't tear down the other resources.

I left it in from the branch I continued off of, but I believe it will require some more testing and configuration. Left it alone for now since `macie` `firewall-manager` and `compliance` (`security-hub` and `guard-duty`) all share a similar setup of 

Root AWS Account: delegates admin account (`security` aws account)
Security AWS Account: Enables AWS Service and sets up member accounts for all other accounts (non security / root)
Member Account: Scans for security vulnerabilities.

The problem with this structure seems to be from the aws API requiring different accounts to be assumed for setting up this structure

Reply:
it's better to name the var `account_status` and make it `string` since it's used directly in the `status` of `resource "aws_macie2_account"`